Source/inv.cpp

```diff
@@ -1855,14 +1855,34 @@ int InvPutItem(Player &player, Point position)
 	assert(CanPut(position));
 
 	int ii = AllocateItem();
+	auto &item = Items[ii];

```

Comment:
The rest of this procedure is now functionally identical to SyncDropItem (except for the NewCursor call which really should be moved out) - could dedupe the logic by calling SyncDropItem directly.